lib/modules/measure/MeasureService.ts

```diff
@@ -444,3 +448,61 @@ export class MeasureService {
     return measureName.asset;
   }
 }
+
+/**
+ * Create a new document in the collection asset-history, for each asset states
+ */
+async function historizeAssetStates(
+  assetStates: Map<number, KDocument<AssetContent<any, any>>>,
+  engineId: string,
+  originalAssetMetadata: Metadata,
+  assetMetadata: Metadata
+): Promise<void> {
+  const metadataChanges = objectDiff(originalAssetMetadata, assetMetadata);
+  const lastTimestampRecorded = Array.from(assetStates.keys()).pop();
+
+  const histories: AssetHistoryContent[] = [];
+  for (const [measuredAt, assetState] of assetStates) {
```

Comment:
Isn't it this?
```suggestion
  for (const [measuredAt, assetState] of assetStates.entries()) {
```

Reply:
[Since `Map` is iterable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#iterating_map_with_for...of) you can do it both ways:
```ts
for (const [key, value] of myMap);
// AND
for (const [key, value] of myMap.entries());
```

Reply:
Oh ok I didn't knew. You can leave it like this :+1: